runtime/arm64.S

```diff
@@ -185,6 +185,21 @@ G(name):
         .size   G(name), .-G(name)
 #endif
 
+/* Function prologue and epilogue */
+
+.macro ENTER_FUNCTION
+        CFI_OFFSET(29, -16)
+        CFI_OFFSET(30, -8)
+        stp     x29, x30, [sp, -16]!
+        CFI_ADJUST(16)
+        add     x29, sp, #0
+.endm
+
+.macro LEAVE_FUNCTION
+        ldp     x29, x30, [sp], 16
+        CFI_ADJUST(-16)
+.endm
+
```

Comment:
On the ordering of CFI directives, should they be before or after the instruction they relate to? When I compile C code to get the assembly, I get the instruction first and then the CFI directive. For compiled OCaml code and runtime assembly there is a mix of before and after. Does it matter? For example on entry to a main function:

```
_main:                                  ; @main
	.cfi_startproc
; %bb.0:
	sub	sp, sp, #64
	.cfi_def_cfa_offset 64
	stp	x29, x30, [sp, #48]             ; 16-byte Folded Spill
	add	x29, sp, #48
	.cfi_def_cfa w29, 16
	.cfi_offset w30, -8
	.cfi_offset w29, -16
   ...
```

Reply:
My understanding is that CFI annotations must appear after the instruction which causes the changes they refer to.